nrf_security/doc/drivers.rst

```diff
@@ -0,0 +1,115 @@
+
+The Arm CryptoCell cc3xx hardware is initialized in :file:`<NCS>/nrf/drivers/hw_cc310/hw_cc310.c` and is controlled with the :kconfig:`CONFIG_HW_CC3XX` Kconfig variable.
+The Kconfig variable has a default value of 'y' when cc3xx is available in the SoC.
+
+.. _nrf_security_driver_oberon:
+
+nrf_oberon driver
+*****************
+
+The :ref:`nrf_oberon_readme` is distributed as a closed-source binary that provides select cryptographic algorithms optimized for use in nRF devices.
+This provides faster execution than the original Mbed TLS implementation.
+
+The nrf_oberon driver provides support for AES ciphers, SHA-1, SHA-256, SHA-384, SHA-512, ECDH and ECDSA using NIST curve secp224r1 and secp256r1, and ECJPAKE using NIST curve secp256r1.
+
+Enabling the nrf_oberon driver
+==============================
+
+The :ref:`nrf_oberon_readme` PSA driver can be enabled by setting the :kconfig:`PSA_CRYPTO_DRIVER_OBERON` Kconfig variable.
+
+.. _nrf_security_drivers_builtin:
+
+Builtin Mbed TLS
+****************
+
+Builtin Mbed TLS uses the open-source software from the `Arm Mbed TLS project`_ with minimal adjustments in PSA Core and PSA driver scope to support multiple drivers.
+
+The fork of `Arm Mbed TLS project`_ used by nrf_security is checked out using ``west`` and is located in the folder called ``mbedtls`` in the root of nRF Connect SDK.
+
+Builtin Mbed TLS support can be used to add support for features not available in the `Arm CryptoCell cc3xx driver`_, for example AES-256 or ECC Brainpool curve types.
+
+Similarly, you can use the builtin Mbed TLS to add support for features not available in the `nrf_oberon driver`_, for example to add support for more ECC curve types.
+
+.. note:
+   nrf_security provides a replacement to the file ``psa_crypto_driver_wrappers.c`` with adjustments to support multiple drivers.
+
+
+Enabling the builtin Mbed TLS support
+=====================================
+
+To enable the builtin Mbed TLS support, set the :kconfig:`PSA_CRYPTO_DRIVER_BUILTIN` Kconfig variable to true.
+
+It is generally not needed to enable builtin Mbed TLS support manually, as there is Kconfig logic that does that, depending on the enabled cryptographic features or algorithms and the configuration of PSA drivers.
+
+Legacy Mbed TLS
+***************
+
+Some legacy Mbed TLS APIs are still supported, for instance for TLS and DTLS support and backwards compatibility.
+
+Enabling legacy APIs requires enabling one of the available PSA drivers.
+
+.. note::
+   * The legacy Mbed TLS APIs no longer support glued functionality.
+   * Legacy configurations no longer have an effect on the configurations for the secure image of a TF-M build.
+
+Enabling legacy Mbed TLS support
+================================
+
+The legacy Mbed TLS APIs can be configured by setting the option kconfig:`NORDIC_SECURITY_BACKEND` instead of setting the option kconfig:`NRF_SECURITY`.
```

Comment:
```suggestion
You can configure the legacy Mbed TLS APIs by setting the :kconfig:`NORDIC_SECURITY_BACKEND` Kconfig options instead of setting :kconfig:`NRF_SECURITY`.
```
`CONFIG_` before both names?

Reply:
Outdated proposal